node/core/pvf/common/src/worker/mod.rs

```diff
@@ -317,4 +317,85 @@ pub mod thread {
 			Some(*result.0)
 		}
 	}
+
+	#[cfg(test)]
+	mod tests {
+		use super::*;
+
+		#[test]
+		fn get_condvar_should_be_pending() {
+			let condvar = get_condvar();
+			let outcome = *condvar.0.lock().unwrap();
+			assert!(outcome.is_pending());
+		}
+
+		#[test]
+		fn wait_for_threads_with_timeout_return_none_on_time_out() {
+			let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+			let outcome = wait_for_threads_with_timeout(&condvar, Duration::from_secs(1));
```

Comment:
Timeout this big is unnecessary for this test, you can safely reduce it to something like 100ms